Implement feature - exclude table

bingluen opened the pull request:
Add query condition to exclude special table.

bingluen committed:
Add test case for exclude table

bingluen committed:
Implement exclude table feature

bingluen committed:
Update README.md

bingluen committed:
Make test case more readable

youchenlee commented on src/routes.php:
```diff
@@ -1,7 +1,13 @@
+        $exclude = [];
+
+        if ($request->exclude !== null) {
+            $exclude = explode(',', strtolower($request->exclude));
```
why with `strtolower`?

bingluen replied:
The query result of `show table` is lowercase.

I think that table should be exclude which table name same spelling but different case.

That why I call `strtolower`.

youchenlee replied:
might relate to MySQL `lower_case_table_names` config. 
the users should have the control.

bingluen replied:
Thanks for your reminder. I got it. I forgot to consider this configuration of MySQL.
I have removed `strtolower` on my last commit.

youchenlee commented on src/Parser.php:
```diff
@@ -7,17 +7,21 @@
-                'indices' => static::getIndices($table),
-            ];
-        });
+                return [
```
unexpected changes

bingluen replied:
I try to change coding style to prevent this on my least commit a9e95d3

youchenlee commented on src/Parser.php:
```diff
@@ -7,17 +7,21 @@
     {
-        return collect(DB::select('show tables'))->map(function ($row) {
-            $table = head($row);
+        return collect(DB::select('show tables'))
```
Please add one more `map` to prevent the redundant and implicit `head($row)`  statements.

youchenlee commented on tests/ParserTest.php:
```diff
@@ -146,4 +146,33 @@ public function testIndexKey()
+
+        DB::statement('DROP TABLE `should_on_result`');
+    }
 }
```
`exclude=table1,table2` is not covered by the test.

youchenlee commented on tests/ParserTest.php:
```diff
@@ -146,4 +146,33 @@ public function testIndexKey()
+        $expect = ['should_on_result'];
+
+        $schema = Parser::getSchema($exclude);
+        $tables = $schema->map(function ($table) {
```
perhaps `pluck()`

youchenlee reviewed (commented):
I have left some comments inline. Thanks.

bingluen committed:
Improve test case

Add case - exclude multiple table

bingluen committed:
Modify implement detail

1. Add map function to prevent the redundant implicit head($row) statements
2. Change coding style to prevent unexpected changes

bingluen committed:
Remove `strtolower`
